luxonis_ml/nn_archive/config_building_blocks/base_models/input.py

```diff
@@ -65,6 +68,42 @@ class Input(CustomBaseModel):
         max_length=5,
         description="Shape of the input data as a list of integers (e.g. [H,W], [H,W,C], [BS,H,W,C], ...).",
     )
+    layout: str = Field(
+        description="Lettercode interpretation of the input layout (e.g., 'HCHW').",
+        default="NCHW",
+    )
     preprocessing: PreprocessingBlock = Field(
         description="Preprocessing steps applied to the input data."
     )
+
+    @model_validator(mode="before")
+    def validate_layout(
+        cls,
+        values,
+    ):
+        if "layout" in values.keys():
+            values["layout"] = values["layout"].upper()
+
+            if len(values["layout"]) != len(values["shape"]):
+                raise ValueError("Layout and shape must have the same length.")
+
+            if values["layout"][0] != "N":
+                raise ValueError(
+                    "First letter of layout must always be N (batch size)."
+                )
+
+            if values["input_type"] == InputType.IMAGE.value:
+                if "C" not in values["layout"]:
+                    raise ValueError(
+                        "C letter must be present in layout for image input type."
+                    )
+                idx_c = values["layout"].find("C")
+                if values["shape"][idx_c] not in [1, 3]:
```

Comment:
I think we should remove this (1,3) check as well as you might have different inputs also for downstream models that operate on features.

Reply:
Fixed in [c79c200](https://github.com/luxonis/luxonis-ml/pull/109/commits/c79c200fcdac9fa75d8aa969a94a64f368ccca15). Thanks!